spacy/pipeline/transition_parser.pyx

```diff
@@ -729,9 +729,8 @@ class Parser(TrainablePipe):
                     action.do(state.c, action.label)
                     if state.is_final():
                         break
-                if moves.has_gold(eg, start_state.B(0), state.B(0)):
-                    states.append(start_state)
-                    golds.append(gold)
+                states.append(start_state)
+                golds.append(gold)
```

Comment:
I am not 100% sure if this is what we want, so this definitely needs extra sanity checking.